cfme/tests/configure/test_access_control.py

```diff
@@ -130,6 +130,37 @@ def test_user_assign_multiple_groups(appliance, request, group_collection):
             .format(user, view.group_names, group_names))
 
 
+@pytest.mark.tier(2)
+@pytest.mark.uncollectif(lambda appliance: appliance.version < '5.9')
+def test_user_change_groups(appliance, request, group_collection):
+    """Assign a user to multiple groups and confirm that the user can successfully change groups
+    """
+    group_names = [
+        'EvmGroup-super_administrator', 'EvmGroup-administrator', 'EvmGroup-approver',
+        'EvmGroup-auditor', 'EvmGroup-desktop', 'EvmGroup-operator',
+        'EvmGroup-security', 'EvmGroup-user', 'EvmGroup-vm_user', ]
+
+    group_list = [group_collection.instantiate(description=name) for name in group_names]
+
+    user = new_user(appliance, groups=group_list)
+    request.addfinalizer(user.delete)
+    request.addfinalizer(user.appliance.server.login_admin)
+
+    view = appliance.server.login(user)
```

Comment:
I think this should be reworked to use a context here, which will control the login and ensure a return to default appliance user (admin) when the context exits.

```python
with custom_user:
    # testing done while user is logged in
    # changing groups, etc

# out of the context
# now I can login_admin and do more test - or NOT, and the next test won't accidentally be logged in as 'custom_user'
```

Used in remote auth tests: https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/integration/test_cfme_auth.py#L120

Reply:
Here are the methods that control the context manager: 
https://github.com/ManageIQ/integration_tests/blob/c57379fe6213dda6deebcce2b7f5c1f88aceabc5/cfme/configure/access_control/__init__.py#L140

So you can logout/login, do whatever testing you want on the appliance, and within that context it will always use the custom_user instead of admin. Once the contextual code block is complete, `__exit__` will restore the previously assigned user (admin) as the default.

Reply:
Updated to use context. I can't remember why I used request over context other than deleting the user at the end which isn't really required for testing at the moment